enterprise_gateway/services/processproxies/spark_operator.py

```diff
@@ -22,3 +28,61 @@ def __init__(self, kernel_manager: RemoteKernelManager, proxy_config: dict):
         self.group = "sparkoperator.k8s.io"
         self.version = "v1beta2"
         self.plural = "sparkapplications"
+
+    def get_initial_states(self) -> set:
+        """Return list of states in lowercase indicating container is starting (includes running)."""
+        return ["submitted", "pending", "running"]
+
+    def get_container_status(self, iteration: int | None) -> str:
+        """Determines if container is still active.
+
+        Submitting a new kernel application to the spark operator
+        will take a while to be Running and can also fail before it
+        can actually start any pods. So we will validate that the
+        application succeeds on the Spark Operator side and then
+        delegate to parent to check if the application pod is running.
+
+        Returns
+        -------
+        Empty string if the container cannot be found otherwise.
+        The pod application status in case of success on Spark Operator side
+        Or the retrieved spark operator submission status in other cases (e.g. Failed)
+        """
+        with suppress(Exception):
+            custom_resource = client.CustomObjectsApi().get_namespaced_custom_object(
+                self.group,
+                self.version,
+                self.kernel_namespace,
+                self.plural,
+                self.kernel_resource_name,
+            )
+
+            if not custom_resource:
+                return ""
+
+            application_state = custom_resource['status']['applicationState']['state'].lower()
+
+            self.log.debug(f"Checking CRD status: {application_state}")
```

Comment:
This message is being produced every 3 seconds - please make this conditional on `iteration is not None` (if we're keeping this method).